### ISSUE TITLE
`WC_REMOVE_ALL_DATA` constant is not deleting all Tables

### ISSUE THREAD

**Issue by up1512001** [woothemes/woocommerce#44769]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I have defined `WC_REMOVE_ALL_DATA` constant to `wp-config.php` file but its not deleting below four tables.

1. wp_actionscheduler_actions
2. wp_actionscheduler_claims
3. wp_actionscheduler_groups
4. wp_actionscheduler_logs

https://github.com/woocommerce/woocommerce/assets/75293077/67748c1a-98d5-4885-881c-3909e7d2ab11

## Error Logs
```
Deletion failed: .wp-die-message { display: none; } Fatal error: Uncaught Error: Class &quot;WC_Plugin_Updates&quot; not foundin /Users/utsav/Local Sites/woocomm-testing/app/public/wp-content/plugins/woocommerce/includes/wc-core-functions.php on line 2255Call stack:wc_enable_wc_plugin_headers()wp-includes/class-wp-hook.php:324WP_Hook::apply_filters()wp-includes/plugin.php:205apply_filters()wp-includes/functions.php:6824get_file_data()wp-admin/includes/plugin.php:91get_plugin_data()wp-admin/includes/plugin.php:339get_plugins()wp-content/plugins/woocommerce/includes/admin/helper/class-wc-helper.php:1393WC_Helper::get_local_woo_plugins()wp-content/plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php:183WC_Helper_Updater::get_update_data()wp-content/plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php:40WC_Helper_Updater::transient_update_plugins()wp-includes/class-wp-hook.php:326WP_Hook::apply_filters()wp-includes/plugin.php:205apply_filters()wp-includes/option.php:2360set_site_transient()wp-admin/includes/plugin.php:1031delete_plugins()wp-admin/includes/ajax-actions.php:4748wp_ajax_delete_plugin()wp-includes/class-wp-hook.php:324WP_Hook::apply_filters()wp-includes/class-wp-hook.php:348WP_Hook::do_action()wp-includes/plugin.php:517do_action()wp-admin/admin-ajax.php:188Query MonitorThere has been a critical error on this website.Learn more about troubleshooting WordPress.
```

### Expected behavior

It should delete all tables created by WooCommerce Plugin.

### Actual behavior

Its not deleting all tables created by WooCommerce Plugin.

### Steps to reproduce

1. Create a blank Local WP
2. Install Query Monitor and WooCommerce Plugin
3. Define `WC_REMOVE_ALL_DATA` to be true in `wp-config.php`
4. deactivate WooCommerce then delete it.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): http://woocomm-testing.local
Site address (URL): http://woocomm-testing.local
WC Version: 8.6.0
REST API Version: ✔ 8.6.0
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.1
Log Directory Writable: ✔
WP Version: 6.4.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.3.0
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 8.4.0
(SecureTransport) LibreSSL/3.3.6

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.6.0
WC Database Prefix: wp_
Total Database Size: 2.66MB
Database Data Size: 1.00MB
Database Index Size: 1.66MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 2
page: 8
post: 2
revision: 4
wp_template: 2

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (2) ###

Query Monitor: by John Blackbourn – 3.15.0
WooCommerce: by Automattic – 8.6.0

### Dropin Plugins (1) ###

db.php: Query Monitor Database Class (Drop-in)

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### WC Pages ###

Shop base: #18 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.0
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /Users/utsav/Local Sites/woocomm-testing/app/public/wp-content/plugins/woocommerce/templates/block-notices/error.php
/Users/utsav/Local Sites/woocomm-testing/app/public/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/Users/utsav/Local Sites/woocomm-testing/app/public/wp-content/plugins/woocommerce/templates/block-notices/success.php

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-pre-publish-modal
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-02-20 12:07:39 +00:00
Options: ✔
Notes: 1
Onboarding: -

### Action Scheduler ###

Complete: 2
Oldest: 2024-02-19 12:17:50 +0000
Newest: 2024-02-19 12:17:50 +0000

Pending: 2
Oldest: 2024-02-19 12:18:50 +0000
Newest: 2024-02-20 12:17:50 +0000

### Status report information ###

Generated at: 2024-02-19 12:18:04 +00:00
`

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by barryhughes** [comment]
The Action Scheduler tables are a special case, because:

- Many plugins ship Action Scheduler, not just WooCommerce.
- Only one instance is in memory at any one time.
- The loaded instance has no knowledge of other copies of Action Scheduler.

So, automatically removing these tables is risky and could impact other plugins. What we can certainly do, though, is document this a little better and provide some notes on resolving this manually (or via WP CLI).

**Comment by jorgeatorres** [comment]
As explained, we can't really remove the A-S tables safely when uninstalling WooCommerce as they might be in use by other plugins or extensions. I've logged an issue in the Action Scheduler repo (https://github.com/woocommerce/action-scheduler/issues/1264) to add a CLI command that would allow this, but will close this one as it's not necessarily a WooCommerce problem (same thing would happen with other plugins using A-S).

If I've missed anything, feel free to comment or re-open. Thanks!
